tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -727,17 +727,30 @@ const DeviceCommand EnqueueWrapCommand::assemble_device_command(uint32_t) {
 }
 
 void EnqueueWrapCommand::process() {
-    uint32_t write_ptr = this->manager.get_issue_queue_write_ptr();
-    uint32_t space_left = DeviceCommand::COMMAND_ISSUE_REGION_SIZE - write_ptr;
+    if (this->wrap_region == DeviceCommand::WrapRegion::COMPLETION) {
+        DeviceCommand cmd;
+        cmd.wrap(this->wrap_region);
+        uint32_t write_ptr = this->manager.get_issue_queue_write_ptr();
+        uint32_t cmd_size = DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND;
+        this->manager.cq_reserve_back(cmd_size);
+        this->manager.cq_write(cmd.get_desc().data(), DeviceCommand::NUM_BYTES_IN_DEVICE_COMMAND, write_ptr);
+        this->manager.cq_push_back(cmd_size);
+    } else {
+
+        // TODO: why does this fill the entire available space???
+
+        uint32_t write_ptr = this->manager.get_issue_queue_write_ptr();
+        uint32_t space_left = DeviceCommand::COMMAND_ISSUE_REGION_SIZE - write_ptr;
 
-    // Since all of the values will be 0, this will be equivalent to
-    // a bunch of NOPs
-    vector<uint32_t> command_vector(space_left / sizeof(uint32_t), 0);
```

Comment:
@DrJessop why can't we send the wrap command similar to the case for wrapping the completion? 

BTW I would like to consolidate these two paths but just pushed this to fix a hang I was hitting